core/src/main/java/hudson/model/Queue.java

```diff
@@ -1348,6 +1348,26 @@ public static boolean tryWithLock(Runnable runnable) {
             return queue._tryWithLock(runnable);
         }
     }
+
+    /**
+     * Invokes the supplied {@link Runnable} if the {@link Queue} lock was obtained within the given timeout.
+     *
+     * @param runnable the operation to perform.
+     * @return {@code true} if the lock was acquired within the timeout and the operation was performed.
+     * @since TODO
+     */
+    public static boolean tryWithLock(Runnable runnable, long timeout, TimeUnit unit) throws InterruptedException {
```

Comment:
Would it be simpler to take a `Duration`?

```java
tryWithLock(()=> ..., Duration.ofSeconds(1));
```


```suggestion
    public static boolean tryWithLock(Runnable runnable, Duration timeout) throws InterruptedException {
```

Reply:
Oh yea, I was matching the [JDK signature](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/util/concurrent/locks/Lock.html#tryLock(long,java.util.concurrent.TimeUnit)) of `Lock.tryLock(long time, TimeUnit unit)`, but that's a signature which predates Duration and there's no reason for us not to use Duration here. I'll push an update with that. 👍